src/silo/config/config_repository.cpp

```diff
@@ -0,0 +1,51 @@
+#include "silo/config/config_repository.h"
+
+#include <map>
+#include <string>
+
+#include "silo/config/config_exception.h"
+#include "silo/config/database_config_reader.h"
+
+namespace silo {
+
+ConfigRepository::ConfigRepository(const silo::DatabaseConfigReader& reader)
+    : reader_(reader) {}
+
+DatabaseConfig ConfigRepository::readConfig(const std::filesystem::path& config_path) const {
+   const auto config = reader_.readConfig(config_path);
+   validateConfig(config);
+   return config;
+}
+
+void ConfigRepository::validateConfig(const DatabaseConfig& config) const {
+   std::map<std::string, DatabaseMetadataType> metadata_map;
+   for (const auto& metadata : config.schema.metadata) {
+      if (metadata_map.find(metadata.name) != metadata_map.end()) {
+         throw ConfigException("Metadata name is not unique: " + metadata.name);
```

Comment:
I think that makes it a bit clearer what the problem was:
```suggestion
         throw ConfigException("Metadata name is defined twice in the config:  " + metadata.name);
```